app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -774,8 +794,22 @@ public void setBlockingEnabled(boolean enabled) {
             webView.setBlockingEnabled(enabled);
         }
 
-        backgroundView.setBackgroundResource(enabled ? R.drawable.animated_background : R.drawable.animated_background_disabled);
-        backgroundTransition = (TransitionDrawable) backgroundView.getBackground();
+        final int backgroundResource = enabled ? R.drawable.animated_background : R.drawable.animated_background_disabled;
+        statusBarBackgroundView.setBackgroundResource(backgroundResource);
+        appBarLayout.setBackgroundResource(backgroundResource);
+        backgroundTransitionGroup = getUpdatedBackgroundTransitionGroup(); // must be called after resource update.
+    }
+
+    /**
+     * Returns an updated drawable group for the background views, based on their current view resources.
+     *
+     * For more details on these background views, see comments around their declaration in fragment_browser.xml.
+     */
+    private TransitionDrawableGroup getUpdatedBackgroundTransitionGroup() {
```

Comment:
Nit: I might call this getCurrentBackgroundTransitionGroup - "updated" from what?